### ISSUE TITLE
brew install osgeo/osgeo4mac/gdal2-mrsid fails

### ISSUE THREAD

**Issue by Kollensperger** [OSGeo/homebrew-osgeo4mac#250]
brew install osgeo/osgeo4mac/gdal2-mrsid fails with the following error:
Error: Operation not permitted - /usr/local/lib/gdalplugins/2.2

as far as I can read form other similar issues, this may be related to permissions?
I am on OS X Sierra 10.12.6

brew doctor reports 

> Please note that these warnings are just used to help the Homebrew maintainers
> with debugging if you file an issue. If everything you use Homebrew for is
> working fine: please don't worry and just ignore them. Thanks!
> 
> Warning: Anaconda is known to frequently break Homebrew builds, including Vim and
> MacVim, due to bundling many duplicates of system and Homebrew-available
> tools.
> 
> If you encounter a build failure please temporarily remove Anaconda
> from your $PATH and attempt the build again prior to reporting the
> failure to us. Thanks!
> 
> Warning: python is symlinked to python3
> This will confuse build scripts and in general lead to subtle breakage.
> 
> Warning: "config" scripts exist outside your system or Homebrew directories.
> `./configure` scripts often look for *-config scripts to determine if
> software packages are installed, and what additional flags to use when
> compiling and linking.
> 
> Having additional scripts in your path can confuse software installed via
> Homebrew if the config script overrides a system or Homebrew provided
> script of the same name. We found the following "config" scripts:
>   /Users/peter/anaconda/bin/curl-config
>   /Users/peter/anaconda/bin/freetype-config
>   /Users/peter/anaconda/bin/glue-config
>   /Users/peter/anaconda/bin/icu-config
>   /Users/peter/anaconda/bin/libpng-config
>   /Users/peter/anaconda/bin/libpng16-config
>   /Users/peter/anaconda/bin/python3-config
>   /Users/peter/anaconda/bin/python3.6-config
>   /Users/peter/anaconda/bin/python3.6m-config
>   /Users/peter/anaconda/bin/xml2-config
>   /Users/peter/anaconda/bin/xslt-config
>

**Comment by pat-s** [comment]
did you already successfully install `mrsid-sdk`? 

Thats where I am currently stuck trying to reproduce your problem. trying do manually download the SDK atm

```ruby
Installing osgeo/osgeo4mac/gdal2-mrsid dependency: mrsid-sdk
Error: Formula expects to locate the following archive:
  MrSID_DSDK-9.5.1.4427-darwin14.universal.clang60.tar.gz

in the HOMEBREW_CACHE directory:
  /Users/pjs/Library/Caches/Homebrew

Copy archive to cache or create a symlink in cache
```

**Comment by Kollensperger** [comment]
Yes I've managed to install mrsid-sdk. The problem was that osgeo is looking for a deprecated version that is found on the lizardtech website. In addition the iOS and Mac versions might be bungled up. I downloaded all of them and the moved exactly the one homebrew was looking for to the folder

**Comment by pat-s** [comment]
Now I can reproduce your issue. 
However, I cannot help here although I looked at the formula :(

You need to wait for a reply of @dakcarto. 

`mrsid-sdk` formula should also be updated so that users do not need to grab the deprecated SDK via a [hidden url](http://bin.lizardtech.com/download/developer/MrSID_DSDK-9.5.1.4427-darwin14.universal.clang60.tar.gz).

**Comment by Kollensperger** [comment]
Thanks for looking Patrick, I've also tried on Linux but whilst I can get gdal with mr Sid running in command line there, I can't get it running on qgis which insists on a later version of gdal. 
Hopefully a solution will be found soon.

**Comment by jerr0328** [comment]
Interesting, I tried to update the gdal2 formula and got basically the same error in Travis. Not sure what to do at this point.

**Comment by spolloni** [comment]
Also experiencing the same issue after after trying to reinstall qgis2

**Comment by spolloni** [comment]
In my case, this error has been arising for a subset of formulas required by qgis2, including gdal2, qt4 and sip-qt4.  Installing with `brew install -verbose` gives a little more detail on what's going on:

> sandboxd[79052]: ruby(92586) deny file-write-create /usr/local/share/sip
> Error: Operation not permitted - /usr/local/share/sip

So this does seem like a permission issue. However, all my attempts have failed to remedy the situation. Some of these are borderline irresponsible: 

1.  Giving permission with `sudo chown -R $(whoami) /usr/local`.
2.  Disabling the OS's _System Integrity Protection_ with `csrutil disable` while booting in Recovery Mode. 
3.  Messing with homebrew's executables (brew.sh) to allow for `sudo brew install `.
4. Doing a fresh install of homebrew in my $HOME directory and trying to install the formulas from there. 

This is really mind-boggling to me. In the case of sip-qt4, for example, the error arises because the script tries to create the /usr/local/share/sip directory. I am able to manually `mkdir /usr/local/share/sip` right after the error, and trying to install again will prompt the same error when the script tries to create yet another directory. 

I am 100% sure that the permissions are properly set, using `ls -l`. Moreover, I managed to successfully  install some of these problematic formulas just a few months ago with the same OS.

**Comment by spolloni** [comment]
@Kollensperger @jerr0328  https://github.com/Homebrew/homebrew-core/issues/16443

**Comment by dakcarto** [comment]
Hi @Kollensperger, thanks for reporting! Also, thanks to all who worked on this issue.

Should be fixed in d2deb6e.

Note: the MrSID SDK will stay at its current version until the next release of macOS, whereby Mavericks support can be dropped. In the meantime, you can access the older SDKs on the LizardTech dev portal under the SDK section -> `Deprecated versions` link, once you are logged in. Though that page appears to work without a login, it would be inappropriate to post it here.

Once Mavericks support is dropped, we can bump up to another lowest-available version, e.g. `MrSID_DSDK-9.5.4.4703-darwin15.universal.clang70`

**Comment by dakcarto** [comment]
Closing. Please reopen if needed.
